flake.nix

```
@@ -0,0 +1,154 @@
+  };
+
+  outputs = {
+    self,
+    nixpkgs,
+    flake-utils,
+    ...
+  }: let
+    golinkVersion =
+      if (self ? shortRev)
+      then self.shortRev
+      else "dev";
+  in
+    {
+      overlay = final: prev: let
+        pkgs = nixpkgs.legacyPackages.${prev.system};
+      in rec {
+        golink = pkgs.buildGo119Module rec {
+          pname = "golink";
+          version = golinkVersion;
+          src = pkgs.nix-gitignore.gitignoreSource [] ./.;
+
+          vendorSha256 = "sha256-2JCGv41htpeL9ZDmaSBH5hYp9DL4nJCBAASi/VbmwwE=";
+        };
+      };
+    }
+    // flake-utils.lib.eachDefaultSystem
+    (system: let
+      pkgs = import nixpkgs {
+        overlays = [self.overlay];
+        inherit system;
+      };
+    in rec {
+      # `nix develop`
+      devShell = pkgs.mkShell {buildInputs = [pkgs.go];};
+
+      # `nix build`
+      packages = with pkgs; {
+        inherit golink;
+      };
+
+      defaultPackage = pkgs.golink;
+
+      # `nix run`
+      apps.golink = flake-utils.lib.mkApp {
+        drv = packages.golink;
+      };
+      defaultApp = apps.golink;
+
+      overlays.default = self.overlay;
+    })
+    // {
+      nixosModules.default = {
+        pkgs,
+        lib,
+        config,
+        ...
+      }: let
+        cfg = config.services.golink;
+      in {
+        options = with lib; {
+          services.golink = {
+            enable = mkEnableOption "Enable golink";
+
+            package = mkOption {
+              type = types.package;
+              description = ''
+                golink package to use
+              '';
+              default = pkgs.golink;
+            };
+
+            dataDir = mkOption {
+              type = types.path;
+              default = "/var/lib/golink";
+              description = "Path to data dir";
+            };
+
+            user = mkOption {
+              type = types.str;
+              default = "golink";
+              description = "User account under which golink runs.";
+            };
+
+            group = mkOption {
+              type = types.str;
+              default = "golink";
+              description = "Group account under which golink runs.";
+            };
+
+            databaseFile = mkOption {
+              type = types.path;
+              default = "/var/lib/golink/golink.db";
+              description = "Path to SQLite database";
+            };
+
+            tailscaleAuthKeyFile = mkOption {
+              type = types.path;
+              description = "Path to file containing the Tailscale Auth Key";
+            };
+
+            verbose = mkOption {
+              type = types.bool;
+              default = false;
+            };
+          };
+        };
+        config = lib.mkIf cfg.enable {
+          users.users."${cfg.user}" = {
+            home = cfg.dataDir;
+            createHome = true;
+            group = "${cfg.group}";
+            isSystemUser = false;
```

isSystemUser is for services like this one

```suggestion
            isSystemUser = true;
```